Python/PRP/GUI/pyGUICheckBoxCtrl.cpp

```diff
@@ -0,0 +1,106 @@
+/* This file is part of HSPlasma.
+*
+* HSPlasma is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* HSPlasma is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with HSPlasma.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <PRP/GUI/pfGUICheckBoxCtrl.h>
+#include "pyGUICheckBoxCtrl.h"
+#include "PRP/GUI/pyGUIControlMod.h"
+#include "PRP/KeyedObject/pyKey.h"
+
+extern "C" {
+
+PY_PLASMA_NEW(GUICheckBoxCtrl, pfGUICheckBoxCtrl)
+
+PY_METHOD_VA(GUICheckBoxCtrl, addAnimKey,
+    "Params: animation key\n"
+    "Add an animation to the checkbox control")
+{
+    pyKey* key;
+    if (!PyArg_ParseTuple(args, "O", &key)) {
+        PyErr_SetString(PyExc_TypeError, "addAnimKey expects a plKey");
+        return NULL;
+    }
+    if (!pyKey_Check((PyObject*)key)) {
+        PyErr_SetString(PyExc_TypeError, "addAnimKey expects a plKey");
+        return NULL;
+    }
+    self->fThis->addAnimKey(*key->fThis);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_VA(GUICheckBoxCtrl, delAnimKey,
+    "Params: idx\n"
+    "Remove an animation from the checkbox control")
+{
+    int idx;
+    if (!PyArg_ParseTuple(args, "i", &idx)) {
+        PyErr_SetString(PyExc_TypeError, "delAnimKey expects an int");
+        return NULL;
+    }
+    self->fThis->delAnimKey(idx);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_NOARGS(GUICheckBoxCtrl, clearAnimKeys,
+    "Remove all animation keys from the checkbox control")
+{
+    self->fThis->clearAnimKeys();
+    Py_RETURN_NONE;
+}
+
+static PyMethodDef pyGUICheckBoxCtrl_Methods[] = {
+    pyGUICheckBoxCtrl_addAnimKey_method,
+    pyGUICheckBoxCtrl_delAnimKey_method,
+    pyGUICheckBoxCtrl_clearAnimKeys_method,
+    PY_METHOD_TERMINATOR
+};
+
+PY_GETSET_GETTER_DECL(GUICheckBoxCtrl, animationKeys) {
```

Comment:
I'm pretty sure there are at least three different ways this is written in the library... :trollface: